test/functional/test_runner.py

```diff
@@ -24,6 +24,10 @@
 import tempfile
 import re
 import logging
+import unittest
+import test_framework.script
```

Comment:
It seems a shame to have to list the modules out multiple times. If you use the `loadTestsFromName()` method, you can avoid having to import the modules:

```python
TEST_FRAMEWORK_MODULES = [
    "script",
    "key",
    "messages",
]
... 
    # Test Framework Tests
    for module in TEST_FRAMEWORK_MODULES:
        suite = unittest.TestLoader().loadTestsFromName("test_framework.{}".format(module))
        unittest.TextTestRunner(verbosity=1).run(suite)
```

If you wanted to be really clever, you could just search all the modules for which ones import `unittest` and then automatically pick them up in the test runner, so they don't need to be listed out at all. I don't think that's necessary in this PR, but you might want to consider it in a follow-up.

Reply:
Ahhh yes this is what I wanted to do! Ran into a bunch of import errors when I tried to auto discover everything, will look into it further.